src/snyk.ts

```diff
@@ -0,0 +1,293 @@
+import { combineLatest, EMPTY, from } from 'rxjs'
+import { distinctUntilChanged, filter, map, switchMap, tap } from 'rxjs/operators'
+import * as sourcegraph from 'sourcegraph'
+import { ApiOptions, getUserOrgs, listAggregatedProjectIssues, listAllProjects, Project, ProjectIssues } from './api'
+
+interface Configuration {
+    'snyk.corsAnywhereUrl'?: string
+    'snyk.apiToken'?: string
+    'snyk.repositoryNamePattern'?: string
+    'snyk.organizationKeyTemplate'?: string
+}
+
+const getConfig = (): Configuration => sourcegraph.configuration.get<Configuration>().value
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    const panelView = sourcegraph.app.createPanelView('snyk.panel')
+    panelView.title = 'Snyk'
+    panelView.content = 'Open a file to see the Snyk report'
+
+    // Don't show multiple alerts for the same project in the same session
+    const shownProjectAlerts = new Set<string>()
+
+    context.subscriptions.add(
+        combineLatest([
+            from(sourcegraph.app.activeWindowChanges).pipe(
+                switchMap(activeWindow => activeWindow?.activeViewComponentChanges || EMPTY),
+                filter((viewer): viewer is sourcegraph.CodeEditor => !!viewer && viewer.type === 'CodeEditor'),
+                distinctUntilChanged((a, b) => a.document === b.document)
+            ),
+            from(sourcegraph.configuration).pipe(map(() => getConfig())),
+        ])
+            .pipe(
+                tap(() => panelView.content = 'Loading...'),
+                switchMap(async ([editor, config]) => {
+                    try {
+                        // Construct API Options
+                        const corsAnywhereUrl = new URL(
+                            (config['snyk.corsAnywhereUrl']?.replace(/\/$/, '') ||
+                                'https://cors-anywhere.herokuapp.com') + '/'
+                        )
+
+                        const apiToken = config['snyk.apiToken']
+                        if (!apiToken) {
+                            throw new NoApiTokenError()
+                        }
+
+                        const snykBaseUrl = new URL('https://snyk.io/')
+                        const apiOptions: ApiOptions = {
+                            snykApiUrl: new URL(
+                                `${corsAnywhereUrl.href.replace(/\/$/, '')}/${snykBaseUrl.href.replace(/\/$/, '')}/`
+                            ),
+                            apiToken,
+                        }
+
+                        panelView.content = 'Loading...'
+                        const uri = new URL(editor.document.uri)
+                        const shortRepoName = decodeURIComponent(uri.pathname).replace(/^\//, '')
+                        const filePath = decodeURIComponent(uri.hash.slice(1))
+
+                        const repositoryNamePattern = '(?:^|/)([^/]+)/([^/]+)$'
```

Comment:
*question* Were you intending to get this value from the setting `snyk.repositoryNamePattern`?

Reply:
Yes, thanks!